src/firmware/app/control/trajectory_planner.c

```diff
@@ -0,0 +1,337 @@
+#include "firmware/app/control/trajectory_planner.h"
+
+#include <assert.h>
+#include <math.h>
+#include <stdbool.h>
+
+#include "assert.h"
+#include "firmware/shared/math/polynomial_1d.h"
+#include "firmware/shared/math/polynomial_2d.h"
+
+// The following function is based on an algorithm described in
+// http://www2.informatik.uni-freiburg.de/~lau/students/Sprunk2008.pdf
```

Comment:
Can you point to a specific algorithm name or page number, since that paper is very long?

Reply:
Added page # :+1: